src/modules/Elsa.MongoDB/Features/MongoDbFeature.cs

```diff
@@ -0,0 +1,117 @@
+using System.Security.Authentication;
+using Elsa.Features.Abstractions;
+using Elsa.Features.Services;
+using Elsa.Http.Models;
+using Elsa.Identity.Contracts;
+using Elsa.Identity.Entities;
+using Elsa.Labels.Contracts;
+using Elsa.Labels.Entities;
+using Elsa.MongoDB.HostedServices;
+using Elsa.MongoDB.Modules.Identity;
+using Elsa.MongoDB.Modules.Labels;
+using Elsa.MongoDB.Modules.Management;
+using Elsa.MongoDB.Modules.Runtime;
+using Elsa.MongoDB.Options;
+using Elsa.MongoDB.Serializers;
+using Elsa.Workflows.Core.Services;
+using Elsa.Workflows.Core.State;
+using Elsa.Workflows.Management.Contracts;
+using Elsa.Workflows.Management.Entities;
+using Elsa.Workflows.Runtime.Contracts;
+using Elsa.Workflows.Runtime.Entities;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using MongoDB.Bson.Serialization;
+using MongoDB.Bson.Serialization.Serializers;
+using MongoDB.Driver;
+using MongoDB.Driver.Core.Extensions.DiagnosticSources;
+using StoredBookmark = Elsa.Workflows.Runtime.Entities.StoredBookmark;
+using WorkflowExecutionLogRecord = Elsa.Workflows.Runtime.Entities.WorkflowExecutionLogRecord;
+
+namespace Elsa.MongoDB.Features;
+
+/// <summary>
+/// Configures MongoDb.
+/// </summary>
+public class MongoDbFeature : FeatureBase
+{
+    /// <inheritdoc />
+    public MongoDbFeature(IModule module) : base(module)
+    {
+    }
+
+    /// <summary>
+    /// A delegate that configures MongoDb.
+    /// </summary>
+    public Action<MongoDbOptions> Options { get; set; } = _ => { };
+
+    /// <inheritdoc />
+    public override void Apply()
+    {
+        Services.Configure(Options);
+
+        Services.AddSingleton(CreateDatabase);
+        
+        RegisterSerializers();
+
+        AddMongoCollection<Application>(Services, "applications");
+        AddMongoCollection<User>(Services, "users");
+        AddMongoCollection<Role>(Services, "roles");
+        AddMongoCollection<Label>(Services, "labels");
+        AddMongoCollection<WorkflowDefinitionLabel>(Services, "workflow_definition_labels");
+        AddMongoCollection<WorkflowDefinition>(Services, "workflow_definitions");
+        AddMongoCollection<WorkflowInstance>(Services, "workflow_instances");
+        AddMongoCollection<WorkflowState>(Services, "workflow_states");
+        AddMongoCollection<WorkflowExecutionLogRecord>(Services, "workflow_execution_logs");
+        AddMongoCollection<StoredTrigger>(Services, "triggers");
+        AddMongoCollection<StoredBookmark>(Services, "bookmarks");
```

Comment:
Can we add these collections supporting different modules in a way such that they are only created when the individual module persistence is enabled?

For instance, my application might not use the Labels module, so I prefer not to have a "labels" collection when I use the MongoDB provider.